libs/particles/effects.ts

```diff
@@ -37,24 +37,30 @@ namespace particles {
             anchor.setFlag(SpriteFlag.Ghost, true);
             this.sourceFactory(anchor, particlesPerSecond);
             anchor.lifespan = lifespan;
-            
-            // clear out some pixels at random to give feedback on sprite being destroyed
-            const replacementImage = anchor.image.clone();
-            if (replacementImage) {
-                const r = new Math.FastRandom();
-                for (let i = (replacementImage.width * replacementImage.height) >> 4; i > 0; --i) {
-                    const x = r.randomRange(0, replacementImage.width)
-                    const y = r.randomRange(0, replacementImage.height)
-                    const w = r.randomRange(1, 3);
-                    const h = r.randomRange(1, 3);
-                    // replacementImage.setPixel(r.randomRange(0, replacementImage.width), r.randomRange(0, replacementImage.height), 0);
-                    replacementImage.drawRect(x, y, w, h, 0);
-                }
-                anchor.setImage(replacementImage);
-            }
+            dissolveSprite(anchor);
         }
     }
 
+    /**
+     * Dissolve the given sprite by removing random sections of it's image
+     * @param sprite
+     */
+    function dissolveSprite(sprite: Sprite) {
```

Comment:
can this be a particle effect?

Reply:
How about as an `image effect` class with the other two I mentioned ('invert' or randomize colors and one layer of the melt effect that occurs on for game over), as there aren't any particles that come out of it. Also makes it more reusable with background images / random images, and not specifically tied to sprites implementation.

also, bonus feed the ducks game:

![ezgif com-optimize 2](https://user-images.githubusercontent.com/5615930/50793409-f52b2180-127c-11e9-8cbb-1f1683a6c7c9.gif)